src/colorizer/CanvasUIOverlay.ts

```diff
@@ -357,18 +387,23 @@ export default class CanvasOverlay {
    * Render the overlay to the canvas.
    */
   render(): void {
-    const ctx = this.canvas.getContext("2d") as OffscreenCanvasRenderingContext2D | null;
+    const ctx = this.canvas.getContext("2d") as CanvasRenderingContext2D | null;
     if (ctx === null) {
       console.error("Could not get canvas context");
       return;
     }
 
-    const devicePixelRatio = window.devicePixelRatio || 1;
+    const devicePixelRatio = this.getPixelRatio();
     this.canvas.width = this.canvasWidth * devicePixelRatio;
     this.canvas.height = this.canvasHeight * devicePixelRatio;
     //Clear canvas
     ctx.clearRect(0, 0, this.canvas.width, this.canvas.height);
 
+    // Render the viewport
+    super.render();
+    ctx.imageSmoothingEnabled = false;
+    ctx.drawImage(super.domElement, 0, 0);
```

Comment:
Yup, that's right! The ColorizeCanvas's output isn't directly visible on the page, and is grabbed and rendered by the overlay here.

Unfortunately we can't keep them as separate canvases. Everything needs to be in one canvas in order for the export functionality to work, since it's grabbing the canvas' current image to make into an exported image/video frame.

Reply:
Ok - my only concern is realtime performance but it's probably not a bad bottleneck at the frame rates we expect.